### ISSUE TITLE
Use of ‘UBFC-rPPG_ UBFC-rPPG_PURE_PHYSFORMER’

### ISSUE THREAD

**Issue by Anser1** [ubicomplab/rPPG-Toolbox#338]
First of all, thank you so much for providing the community with an exciting open source toolkit. When I used 'UBFC-rPPG_ UBFC-rPPG_PURE_PHYSFORMER', I found that the model training results were somewhat unsatisfactory. To be specific, I found that the learning rate and training loss of the model given after the training were almost a constant, and the performance of the final evaluation indicators of the model was not very good. I put the details below. I set specific situations in the profile which I also put in the picture below. I would like to ask you why this is the case.
[The contents of the specific 'yaml' file.txt](https://github.com/user-attachments/files/17947681/The.contents.of.the.specific.yaml.file.txt)
![558944bd41368af7ac44b564b26566e](https://github.com/user-attachments/assets/410e2036-80b0-469e-90d0-6de7c06acdd4)
![c970e179e272e69350823217b283eaa](https://github.com/user-attachments/assets/30c37eed-9e31-48e2-a637-6b19369234ed)

**Comment by yahskapar** [comment]
I'm not exactly sure why, and this may require further experimentation with respect to hyperparameters to begin with, but it's worth noting that training on PURE and testing on UBFC-rPPG works pretty well with PhysFormer, but training on PURE and testing on UBFC-rPPG does not work well with PhysFormer (as you've observed). This is consistent with the [toolbox paper (see table 2)](https://arxiv.org/pdf/2210.00716) as well as numerous recent papers that tried that same train-test data pair: [example 1](https://ieeexplore.ieee.org/iel7/6046/4456689/10430462.pdf?casa_token=uk5F1sHu-D8AAAAA:hIecxv7icoYCmCaLkRMyDcJ1pV4MYByvXcle_flbzCIdWeQXdv1YRQX6SQsN6YwD8Gbk0pxapn0) (see table in the second column on page 12) and [example 2](https://arxiv.org/pdf/2411.01542) (see Table 2). Let me know if you (or anyone else for that matter) finds a counterexample, I'd love to dig into that more.

There could be a number of reasons as to why this happens when it comes to cross or inter-dataset training scenarios like these, including perhaps most obviously differences in dataset characteristics causing issues with generalization from one dataset to another due to the domain shift involved. My guess, after spending a limited amount of time investigating this in the past, is that PhysFormer specifically learns less while training on UBFC-rPPG that results in poor generalization to the PURE dataset. Maybe this is because of a lack of subtle, varied motion in the UBFC-rPPG dataset and lighting differences, or some other reason.

**Comment by yahskapar** [comment]
Closing this issue due to the lack of further discussion, but feel free to re-open or make a new issue in case you come across something interesting on this matter. Thanks!
